waiter/src/waiter/util/http_utils.clj

```diff
@@ -286,3 +286,12 @@
   [user-agent-products request]
   (when-let [user-agent (some-> request (get-in [:headers "user-agent"]) str/lower-case)]
     (some #(str/includes? user-agent %) user-agent-products)))
+
+(defn assoc-response-headers
+  "Attaches provided headers into the response."
+  [response & kvs]
+  (loop [headers (get response :headers)
+         [k v & rem-hvs] kvs]
+    (if (str/blank? k)
```

Comment:
Is this a typo? Did you mean to filter out blank values rather than blank keys? Only asking because the way you use this, it looks like you might end up with blank values, but there's no way the literal strings you pass in could be blank.

```suggestion
    (if (str/blank? v)
```

Reply:
This is intentional, blank keys trigger termination of the loop, perhaps it is better to check as `nil?`